src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -388,11 +388,17 @@
       factory: function(o, n) {
         var key = this.daoKey;
         var X = this.ctrl.__subContext__;
-        // FIXME: change to a better implementation
+
         if ( ! X[key] ) {
           // if DAO doesn't exist in context, change daoKey from localMyDAO
           // (server-side) to myDAO (accessible on front-end)
-          key = key.substring(5, 6).toLowerCase() + key.substring(6);
+          for ( var i = 0 ; i < key.length ; i++ ) {
+            if ( key.charAt(i) == key.charAt(i).toUpperCase() ) {
+              key = key.substring(i);
+              break;
+            }
+          }
+          key = key.charAt(0).toLowerCase() + key.slice(1);
```

Comment:
if we have "afexBusinessDAO", would it become "businessDAO"?

Reply:
Thats a good point, if this dao did not exist in the context with the previous implementation it would return 'usinessDAO' which would be broken entirely. It was basically taking out the 'local' out of any dao not in the context by removing the first 5 characters from the string. I think when this was originally implemented the daoKey not in the context was supposed to always be a local dao, which in the case of the user model it was actually bareUserDAO. I'm open to other suggestions if someone has one.

Reply:
What I could do is check if the daoKey contains 'local' in the string and then perform the formatting, that still would not solve the issue of 'bareUserDAO' not being in the context and having to use 'userDAO' instead. I'm assuming afexBusinessDAO would be in the context, and i'm also assuming 'localUserDAO' is what was expected instead of 'bareUserDAO'.

Reply:
The fact that we have "bareUserDAO" as daoKey in the approval request might because EasyDAO add ApprovableAwareDAO to it. @jlhughes @nanoscott can we skip ApprovableAware on bareUserDAO?

@blakegreen32 yes, you can check for startsWith 'local' then strip it out if still not found in the context then throw.

Reply:
I would prefer to restrict to local (non-local) so test for local. Anything else we'll deal with explicitly.  We are slowly adding local for all daos, so they will be there going forward.

Reply:
and regarding the usersDAOs. localUserDAO should have remained the base, and everything built on top of it, but it's a little late for that.